### PR TITLE
pingpong: improve transaction scheduling

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

Currently the transaction scheduling logic in pingpong is lots of complicated sleep calls. Some sleeps were definitely extra and were hurting tps; this PR deprecates them. The whole scheduling logic is replaced with a much simpler one. Let timestamp sequence `{t}_{i >= 0} = {currenttime + 1 / tps * i}`. Then transaction `i` is sent at time `t_i` with best effort.

In my tests `./pingpong run --tps 115 --rest 0 --refresh 1800 --numaccounts 500` produced 440 txns per block. Now `./pingpong run --tps 115 --refresh 1800 --numaccounts 500` produces 475 txns per block. (`--rest` was deprecated)

Related to https://github.com/algorand/go-algorand-internal/issues/1833.

## Test Plan

I ran it.

### CODE REVIEW

File: shared/pingpong/pingpong.go [hunk 1]
```diff
@@ -249,6 +249,17 @@ func computeAccountMinBalance(client libgoal.Client, cfg PpConfig) (fundingRequi
 	return
 }
 
+// Wait for `*nextSendTime` and update it afterwards.
+func schedule(tps uint64, nextSendTime *time.Time) {
+	dur := time.Until(*nextSendTime)
+	if dur > 0 {
```
Comment (manotoor) [inline shared/pingpong/pingpong.go:255]: should there be a check on this that dur is not a large number

File: shared/pingpong/pingpong.go [hunk 2]
```diff
@@ -249,6 +249,17 @@ func computeAccountMinBalance(client libgoal.Client, cfg PpConfig) (fundingRequi
...
+		time.Sleep(dur)
+	}
+
+	*nextSendTime = nextSendTime.Add(
+		time.Duration((1.0 / float64(tps)) * float64(time.Second)))
```
Comment (brianolson) [inline shared/pingpong/pingpong.go:260]:
```suggestion
		time.Duration(float64(time.Second) / float64(tps)))
```

Comment (brianolson) [reply shared/pingpong/pingpong.go:260]: or better yet, at startup, calculate `period := Second/tps` then later `nextSendTime.Add(period)`

**Review by brianolson — COMMENTED** [review]
the `schedule()` abstraction looks good and cleans up a bunch of code. One tweak suggestion.

File: shared/pingpong/pingpong.go (hunk 2 above)
Comment (tolikzinovyev) [reply shared/pingpong/pingpong.go]: done

**Review by brianolson — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**